typespec-extension/src/code-model-builder.ts

```diff
@@ -566,108 +519,146 @@ export class CodeModelBuilder {
         codeModelClient.apiVersions = [];
         for (const version of this.getFilteredApiVersions(
           this.apiVersion,
-          versioning.getVersions(),
+          versions,
           this.options["service-version-exclude-preview"],
         )) {
           const apiVersion = new ApiVersion();
-          apiVersion.version = version.value;
+          apiVersion.version = version;
           codeModelClient.apiVersions.push(apiVersion);
         }
       }
 
-      // server
+      // client initialization
       let baseUri = "{endpoint}";
-      const servers = getServers(this.program, client.service);
-      if (servers && servers.length === 1 && !this.isArmSynthesizedServer(servers[0])) {
-        baseUri = servers[0].url;
-      }
-      const hostParameters = this.processHost(servers?.length === 1 ? servers[0] : undefined);
-      codeModelClient.addGlobalParameters(hostParameters);
+      let hostParameters: Parameter[] = [];
+      client.initialization.properties.forEach((initializationProperty) => {
+        if (initializationProperty.kind === "endpoint") {
+          let sdkPathParameters: SdkPathParameter[] = [];
+          if (initializationProperty.type.kind === "union") {
+            if (initializationProperty.type.values.length <= 2) {
+              // only get the path parameters from the endpoint whose serverUrl is not {"endpoint"}
```

Comment:
nit: this may not always be path parameters in the template. It could also be domain params like `https://{region}.monitor.azure.com`

Reply:
TCGC name it as `SdkPathParameter`, so I use `path parameters`, I will change to `only get the SdkPathParameters from the endpoint whose serverUrl is not {"endpoint"}`, it could be more clear.